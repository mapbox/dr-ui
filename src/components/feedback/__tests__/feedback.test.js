import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './feedback-test-cases.js';

describe('feedback', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.type.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.type;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
