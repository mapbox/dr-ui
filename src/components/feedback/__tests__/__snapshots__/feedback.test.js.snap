// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feedback Basic renders as expected 1`] = `
<div
  className="dr-ui--feedback"
>
  <div>
    <h2
      className="unprose txt-m txt-bold mb6"
    >
      Was this 
      page
       helpful?
    </h2>
    <button
      className="btn btn--s"
      id="dr-ui--feedback-lnglat-yes"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="btn btn--s ml6"
      id="dr-ui--feedback-lnglat-no"
      onClick={[Function]}
    >
      No
    </button>
  </div>
</div>
`;

exports[`feedback Change type renders as expected 1`] = `
<div
  className="dr-ui--feedback"
>
  <div>
    <h2
      className="unprose txt-m txt-bold mb6"
    >
      Was this 
      section
       helpful?
    </h2>
    <button
      className="btn btn--s"
      id="dr-ui--feedback-page-yes"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="btn btn--s ml6"
      id="dr-ui--feedback-page-no"
      onClick={[Function]}
    >
      No
    </button>
  </div>
</div>
`;
